server/src/main/java/org/elasticsearch/index/shard/IndexShard.java

```diff
@@ -1386,7 +1386,7 @@ public CommitStats commitStats() {
      * @throws AlreadyClosedException if shard is closed
      */
     public SeqNoStats seqNoStats() {
-        return getEngine().getSeqNoStats(replicationTracker.getGlobalCheckpoint());
+        return withEngine(engine -> engine.getSeqNoStats(getLastKnownGlobalCheckpoint()));
```

Comment:
Apart from this one, there's also another couple of places to introduce withEngine for getSeqNoStats, that you can see [here](https://github.com/elastic/elasticsearch/pull/129535/commits/a8960ea0cd7198566db860a9f53d1b4a2069d863), it'd be great to also adopt them to cover all code cases.